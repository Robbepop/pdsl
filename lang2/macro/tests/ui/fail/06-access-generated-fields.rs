#![feature(proc_macro_hygiene)]

use ink_lang2 as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(constructor)]
        fn new(&mut self) {
            let _ = &self.__env;
        }

        #[ink(message)]
        fn noop(&self) {
            let _ = &self.__storage;
        }
    }
}

fn main() {}
