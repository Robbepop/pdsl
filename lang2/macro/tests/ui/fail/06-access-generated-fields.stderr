error[E0616]: field `__env` of struct `noop::__ink_private::__ink_storage::StorageAndEnv` is private
  --> $DIR/06-access-generated-fields.rs:13:22
   |
13 |             let _ = &self.__env;
   |                      ^^^^^^^^^^

error[E0616]: field `__storage` of struct `noop::__ink_private::__ink_storage::StorageAndEnv` is private
  --> $DIR/06-access-generated-fields.rs:18:22
   |
18 |             let _ = &self.__storage;
   |                      ^^^^^^^^^^^^^^
